matsim/src/main/java/org/matsim/core/population/PopulationUtils.java

```diff
@@ -833,8 +833,8 @@ public static Activity createAndAddActivity(Plan plan, String type) {
 		return act ;
 	}
 
-	public static Activity createStageActivityFromCoordLinkIdAndMode( final Coord interactionCoord, final Id<Link> interactionLink, String mode ) {
-		Activity act = createActivityFromCoordAndLinkId(PlanCalcScoreConfigGroup.createStageActivityType(mode), interactionCoord, interactionLink);
+	public static Activity createStageActivityFromCoordLinkIdAndModePrefix(final Coord interactionCoord, final Id<Link> interactionLink, String modePrefix ) {
+		Activity act = createActivityFromCoordAndLinkId(PlanCalcScoreConfigGroup.createStageActivityType(modePrefix), interactionCoord, interactionLink);
```

Comment:
Could stage activities created here be of a different Activity (sub)class that fails on setting start/end time? (e.g. throwing an `UnsupportedOperationException` ?)

Reply:
This is already planned (however, without a clear time horizon):
[https://matsim.atlassian.net/projects/MATSIM/issues/MATSIM-847](https://matsim.atlassian.net/projects/MATSIM/issues/MATSIM-847)

Reply:
I see this pull request as another step into the direction of introducing a PlanElement `Waypoint`